src/main/modules/awilix/index.ts

```diff
@@ -27,21 +21,31 @@ const logger: LoggerInstance = Logger.getLogger('app');
  */
 export class Container {
   public enableFor(app: Application): void {
-    app.locals.container = createContainer({ injectionMode: InjectionMode.CLASSIC }).register({
-      logger: asValue(logger),
-      homeGetController: asValue(new HomeGetController()),
-      termsAndConditionsGetController: asValue(new TermsAndConditionsGetController()),
-      languagePreferenceGetController: asValue(new LanguagePreferenceGetController()),
-      languagePreferencePostController: asValue(new LanguagePreferencePostController(new Form(languagePreferenceForm))),
-      hasMarriageBrokenGetController: asValue(new HasMarriageBrokenGetController()),
-      hasMarriageBrokenPostController: asValue(new HasMarriageBrokenPostController(new Form(hasMarriageBrokenForm))),
-      respondentAddressGetController: asValue(new RespondentAddressGetController()),
-      respondentAddressPostController: asValue(new RespondentAddressPostController(new Form(respondentAddressForm))),
-      marriageCertificateGetController: asValue(new MarriageCertificateGetController()),
-      marriageCertificatePostController: asValue(
-        new MarriageCertificatePostController(new Form(marriageCertificateForm))
-      ),
-      errorController: asClass(ErrorController),
+    // const steps = [];
+    const steps = getSteps().map((step: Step) => {
+      const stepDir = `${__dirname}/../../steps/screen-questions/${step.id}`;
+      const view = `${stepDir}/template.njk`;
+      const { getContent, form } = require(`${stepDir}/content.ts`);
+
+      return {
+        [`${step.id}StepGetController`]: asValue(
+          new GetController(fs.existsSync(view) ? view : `${stepDir}/../template.njk`, getContent(step.title))
+        ),
+        [`${step.id}StepPostController`]: asValue(new PostController(new Form(form))),
+      };
     });
+
+    app.locals.container = createContainer({ injectionMode: InjectionMode.CLASSIC }).register(
+      Object.assign(
```

Comment:
Why?

Reply:
This merges the dynamically created get/post controllers with the existing resolvers.

I've refactored this into a `reduce()` called `combinedStepControllersAndResolvers`, which hopefully makes this clearer 🤞